File: framework/languages/java/samples/java/SupportChat/Server/Support/src/main/java/systems/zlink/samples/supportchat/server/support/infrastructure/zlink/spots/conversationspot/handlers/ConversationIdleTimerHandler.java
package systems.zlink.samples.supportchat.server.support.infrastructure.zlink.spots.conversationspot.handlers;

import systems.zlink.framework.spots.ZLinkSpotTimerHandler;
import systems.zlink.framework.spots.ZLinkTimerTick;
import systems.zlink.samples.supportchat.server.support.infrastructure.zlink.spots.conversationspot.ConversationSpot;

public final class ConversationIdleTimerHandler implements ZLinkSpotTimerHandler<ConversationSpot> {
    @Override
    public void handle(ConversationSpot spot, ZLinkTimerTick tick) {
        spot.checkIdle();
    }
}

File: framework/languages/java/samples/kotlin/SupportChat/Server/Support/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/support/infrastructure/zlink/spots/handlers/ConversationIdleTimerHandler.kt
package systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.handlers

import systems.zlink.framework.kotlin.ZLinkSuspendingSpotTimerHandler
import systems.zlink.framework.spots.ZLinkTimerTick
import systems.zlink.samples.kotlin.supportchat.server.support.infrastructure.zlink.spots.ConversationSpot

class ConversationIdleTimerHandler() : ZLinkSuspendingSpotTimerHandler<ConversationSpot> {
    override suspend fun handle(
        spot: ConversationSpot,
        tick: ZLinkTimerTick,
    ) = run {
        spot.checkIdle()
    }
}
